rosidl_typesupport_c/test/test_message_type_support_dispatch.cpp

```diff
@@ -113,7 +116,11 @@ TEST(TestMessageTypeSupportDispatch, get_handle_function) {
   ASSERT_NE(support_map.data[0], nullptr);
   auto * clib = static_cast<const rcpputils::SharedLibrary *>(support_map.data[0]);
   auto * lib = const_cast<rcpputils::SharedLibrary *>(clib);
+
   ASSERT_NE(lib, nullptr);
+  // c-style assert is used to avoid false positive of clang static analysis because it can't
+  // follow gtest asserts
+  assert(nullptr != lib);
```

Comment:
I believe a somewhat simpler solution here is:

```suggestion
  ASSERT(nullptr != lib);
```

(I'm not against what you are doing here on principal, but I think it would be a little simpler to just stick with the GTest macros if possible).

Same goes for the change below.

Reply:
The issue is that scan-build is not following gtest macros to know that it's impossible for lib to be null. I kept the `ASSERT_NE` in line 120 you'll notice so the assert in 123 is actually impossible to fail and gtest will report the test failure in line 120. But scan-build does acknowledge c-style `assert` macros, so it will correctly identify that lib can't be null in line 124.

As far as I can find, you can annotate that functions won't return, but I couldn't find the same for macros.

Reply:
Just out of curiosity, mind changing `ASSERT_NE` to `ASSERT_TRUE(nullptr != lib)`? It seems like there might be some built in magic for it.

https://reviews.llvm.org/D27773
https://clang.llvm.org/doxygen/GTestChecker_8cpp_source.html

Also found:
https://github.com/google/googletest/issues/734

Reply:
Yep, that was my suggestion as well :).

Reply:
Yes, that seems to work. Fixed